tests/installation/system_role.pm

```diff
@@ -34,48 +34,37 @@ sub change_system_role {
             send_key_until_needlematch "system-role-$system_role-selected", 'spc';    # enable role
         }
         else {
-            if (!check_screen("system-role-$system_role-selected")) {
-                # scroll down in case role is not visible straight away
-                send_key_until_needlematch "system-role-$system_role", 'down', 5, 1;
-                assert_and_click "system-role-$system_role";
-                # after role is selected, we get top of the list shown, so might need to scroll again
-                send_key_until_needlematch "system-role-$system_role-selected", 'down', 5, 1;
-            }
+            assert_and_click "system-role-$system_role";
+            assert_and_click "system-role-$system_role-selected";
         }
     }
     else {
         send_key 'alt-' . $role_hotkey{$system_role};
         assert_screen "system-role-$system_role-selected";
     }
-    # every system role other than default will end up in textmode for SLE
-    # But can be minimalx/lxde/xfce
-    set_var('DESKTOP', 'textmode') unless is_opensuse;
 }
 
 sub assert_system_role {
     # Still initializing the system at this point, can take some time
     # Asserting screen with preselected role
     # Proper default role assertion will be addressed in poo#37504
     # Product might or might not have default selected
-    if (is_opensuse) {
-        assert_screen('before-role-selection', 180);
-        change_system_role(get_var('SYSTEM_ROLE', get_var('DESKTOP')));
+    assert_screen('before-role-selection', 180);
+    if (is_opensuse || (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default'))) {
+        # on opensuse is system role 1:1 with DESKTOP, on SLE change system role if it's defined
+        my $system_role = is_opensuse ? get_var('SYSTEM_ROLE', get_var('DESKTOP')) : get_var('SYSTEM_ROLE');
```

Comment:
let me know if this looks more clear to you
`is_sle ? get_var('SYSTEM_ROLE') : get_var('DESKTOP');`
iiuc this is the same thing. wdyt?

Reply:
It's not same.